### PR TITLE
283 feature docs for curated roles

### LINKED ISSUE #283: [FEATURE] Docs for curated roles
**Is your feature request related to a problem? Please describe.**
Similar to our Easytravel role (https://github.com/Dynatrace/ace-box/tree/dev/user-skel/ansible/roles/app-easytravel ) documentation for other curated roles shall be added.

### PR DESCRIPTION
**Pull request by eemrdog**
From the list below, I have completed the readme files of the roles except for Dashboard. Jenkins is also done but
would like to do a second review. Could you please review the completed ones and send your comments?

app-unguard --ok
app-easytravel --ok
dt-oneagent --ok
dt-activegate --ok
microk8s --ok
monaco --ok
gitea --ok
repository --ok
gitlab --ok
Jenkins -- ok but want to check again
Dashboard --not ok
keptn-cli --ok

### CODE REVIEW

**Comment by eemrdog** [comment]
Could you please review all the roles readme below?

app-unguard --ok
app-easytravel --ok
dt-oneagent --ok
dt-activegate --ok
microk8s --ok
monaco --ok
gitea --ok
repository --ok
gitlab --ok
Jenkins -- ok
Dashboard --ok
keptn-cli --ok

File: user-skel/ansible/roles/app-easytravel/Readme.md [hunk 1]
```diff
@@ -39,20 +30,63 @@ easytravel_headless_replicas: 3 #number of headless loadgen replicas
...
+Dynatrace Configurations List:
+
+    Infrastructure:
+      - "auto-tag/app"
+      - "auto-tag/environment"
+      - "conditional-naming-processgroup/ACE Box - containername.namespace"
+      - "conditional-naming-service/app.environment"
+      - "synthetic-location/ACE-BOX"
+    
+    Easy Travel Aplication Specific:
```
Comment (tobigremmer-dt) [inline user-skel/ansible/roles/app-easytravel/Readme.md]: Minor typo: `Easytravel application specific`

File: user-skel/ansible/roles/dashboard/Readme.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
+# dashboard
+
+This currated role can be used to install Acebox Dashboard on a kubernetes cluster.
```
Comment (tobigremmer-dt) [inline user-skel/ansible/roles/dashboard/Readme.md]: As a convention, would be cool if `ACE-Box` is used.

File: user-skel/ansible/roles/dashboard/Readme.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        description: Staging
+        url: "{{ ingress_protocol }}://simplenodeservice-simplenode-jenkins-staging.{{ ingress_domain }}"
+        - section: demo-quality-gates-jenkins
+        description: Production
+        url: "{{ ingress_protocol }}://simplenodeservice-simplenode-jenkins-production.{{ ingress_domain }}"
+        guides:
+        - description: "Quality Gates, Monitoring as a Service and Monitoring as Code - Demo using Jenkins, Gitea and Cloud Automation"
+        url: "{{ ingress_protocol }}://gitea.{{ ingress_domain }}/demo/quality-gates-jenkins/src/branch/main/demo"
+
+  ```
```
Comment (tobigremmer-dt) [inline user-skel/ansible/roles/dashboard/Readme.md:86]: In addition to `useCases` there is also an option to add `extRefs`. See https://github.com/dynatrace-ace/ace-box-ext-template/blob/main/example_roles/my-use-case/templates/my-use-case-dashboard.yml.j2

File: user-skel/ansible/roles/dt-activegate/Readme.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
+# dt-activegate
+
+This currated role can be used to deploy Dynatrace Synthetic-enabled ActiveGate on the acebox.
```
Comment (tobigremmer-dt) [inline user-skel/ansible/roles/dt-activegate/Readme.md]: As a convention, would be cool if ACE-Box is used.

File: user-skel/ansible/roles/dt-oneagent/Readme.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
+# dt-oneagent
+
+This currated role can be used to deploy Dynatrace Oneagent on the acebox.
```
Comment (tobigremmer-dt) [inline user-skel/ansible/roles/dt-oneagent/Readme.md]: As a convention, would be cool if `ACE-Box` is used.

Comment (tobigremmer-dt) [reply user-skel/ansible/roles/dt-oneagent/Readme.md]: Also, the role is actually deploying a Dynatrace k8s operator (which itself deployed OneAgents). Would be great if this is adjusted.

(Kristof created an issue to clean up naming https://github.com/Dynatrace/ace-box/issues/328 )

File: user-skel/ansible/roles/gitea/Readme.md [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+#### "create-secret" 
+This task creates a secret that includes gitea username and password
+
+```yaml
+- include_role:
+    name: gitea
+    tasks_from: create-secret
+```
+
+#### "source-secret"
```
Comment (tobigremmer-dt) [inline user-skel/ansible/roles/gitea/Readme.md:56]: Can we list the actual variable names? E.g. `gitea_username`, ... . Same for sourcing endpoints.

File: user-skel/ansible/roles/gitlab/Readme.md [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+feature_gitlab: false
+gitlab_helm_chart_version: "6.1.2"
+gitlab_domain: "gitlab.{{ ingress_domain }}"
+gitlab_gcpe_helm_chart_version: "0.2.15"
+gitlab_gcpe_tag: "v0.5.3"
+```
+
+### Other Tasks in the Role
+
+#### "source-endpoints"
```
Comment (tobigremmer-dt) [inline user-skel/ansible/roles/gitlab/Readme.md:42]: Can we list the actual variable names? E.g.`gitlab_internal_endpoint`, ... . Same for other variables.

File: user-skel/ansible/roles/jenkins/Readme.md [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+
+#### "create-token" 
+This task generates the jenkins admin token and api token.
+```yaml
+- include_role:
+    name: jenkins
+    tasks_from: create-token
+```
+
+#### "source-endpoints"
```
Comment (tobigremmer-dt) [inline user-skel/ansible/roles/jenkins/Readme.md:103]: Can we list the actual variable names?

File: user-skel/ansible/roles/jenkins/Readme.md [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+```
+
+### Other Tasks in the Role
+
+#### "template-values-file" 
+This task templates the helm values file depending on your use case requirements. This task has to be executed before the "jenkins" role stated above. 
+
+```yaml
+- set_fact:
+    include_jenkins_value_file: "{{ role_path }}/templates/<use case name>-jobs.yml.j2" # rename with your use case name
```
Comment (tobigremmer-dt) [inline user-skel/ansible/roles/jenkins/Readme.md:62]: Can we add a note what this set_fact does? I.e. it specifies where the Jinja template can be found.

A link to an example template would probably also help, for example https://github.com/Dynatrace/ace-box/blob/dev/user-skel/ansible/roles/demo-quality-gates-jenkins/templates/demo-default-jobs.yml.j2

File: user-skel/ansible/roles/jenkins/Readme.md [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+
+- include_role:
+    name: jenkins
+    tasks_from: template-values-file
+  vars:
+    git_username: "<a git tool user name>" # set your git tool´s user name
```
Comment (tobigremmer-dt) [inline user-skel/ansible/roles/jenkins/Readme.md]: I would make it more clear that `git_XYZ` are just examples and these are the actual variables that are substituted.

File: user-skel/ansible/roles/keptn-cli/Readme.md [hunk 11]
```diff
@@ -0,0 +1,21 @@
+# keptn-cli
+
+This currated role install Keptn CLI on the acebox.
```
Comment (tobigremmer-dt) [inline user-skel/ansible/roles/keptn-cli/Readme.md]: Can we add a link to the official Keptn docs please? ( https://keptn.sh/docs/ )

File: user-skel/ansible/roles/microk8s/Readme.md [hunk 12]
```diff
@@ -0,0 +1,21 @@
+# microk8s
+
+This currated role installs microk8s on the ace-box along with the required addons and packages.
+It also configures kube config to automatically connect to the cluster via kubectl and helm.
```
Comment (tobigremmer-dt) [inline user-skel/ansible/roles/microk8s/Readme.md:4]: Can we please add a link to the official microk8s docs?

File: user-skel/ansible/roles/repository/Readme.md [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+  vars:
+    git_remote: "" # e.g. "gitea" or "gitlab" or "github" or any other git tool
+    git_username: "" # a username to connect to a git tool
+    git_password: "" # a password to connect to a git tool
+    git_endpoint: "http://<git_username>:<git_password>@<gitea_internal_endpoint>" # if the git_remote is gitea or gitlab, git_endpoint is generated automatically. For the rest, you can directly set an endpoint
+
+    repo_src: "{{ item.repo_src }}" # the path you want to upload to a git repository
+    git_org: "{{ item.git_org }}" # organization or group name to be created in a git tool
+    git_repo: "{{ item.repo_target }}" # repository name
+  loop:
```
Comment (tobigremmer-dt) [inline user-skel/ansible/roles/repository/Readme.md]: Maybe `loop` overcomplicates things here. For documentation purposes, can we simply show how these variables are usually provided?

File: user-skel/ansible/roles/monaco/Readme.md [hunk 14]
```diff
@@ -0,0 +1,41 @@
+# monaco
+
+This currated role can be used to install monaco CLI on the ace-box.
```
Comment (tobigremmer-dt) [inline user-skel/ansible/roles/monaco/Readme.md]: Can we please add to the official Monaco docs?

**Review by tobigremmer-dt — CHANGES_REQUESTED** [review]
Awesome! Overall, pretty clear without being too detailed and hard to digest. Like the structure.

I flagged some minors.

File: user-skel/ansible/roles/app-easytravel/Readme.md (hunk 1 above)
Comment (eemrdog) [reply user-skel/ansible/roles/app-easytravel/Readme.md]: updated

File: user-skel/ansible/roles/dashboard/Readme.md (hunk 2 above)
Comment (eemrdog) [reply user-skel/ansible/roles/dashboard/Readme.md]: updated

File: user-skel/ansible/roles/dashboard/Readme.md (hunk 3 above)
Comment (eemrdog) [reply user-skel/ansible/roles/dashboard/Readme.md:86]: updated

File: user-skel/ansible/roles/dt-activegate/Readme.md (hunk 4 above)
Comment (eemrdog) [reply user-skel/ansible/roles/dt-activegate/Readme.md]: updated

File: user-skel/ansible/roles/dt-oneagent/Readme.md (hunk 5 above)
Comment (eemrdog) [reply user-skel/ansible/roles/dt-oneagent/Readme.md]: updated changing  the description to k8s operator

File: user-skel/ansible/roles/gitea/Readme.md (hunk 6 above)
Comment (eemrdog) [reply user-skel/ansible/roles/gitea/Readme.md:56]: ok for create-secret. Could not get what you refer to sourcing endpoints? if it is for source-endpoint task, it does not have an input variable as I see

Comment (eemrdog) [reply user-skel/ansible/roles/gitea/Readme.md:56]: I provided "gitea_internal_endpoint" as an output variable from this task. Does it look good?

File: user-skel/ansible/roles/gitlab/Readme.md (hunk 7 above)
Comment (eemrdog) [reply user-skel/ansible/roles/gitlab/Readme.md:42]: added this var as output var from this task

File: user-skel/ansible/roles/jenkins/Readme.md (hunk 8 above)
Comment (eemrdog) [reply user-skel/ansible/roles/jenkins/Readme.md:103]: updated

File: user-skel/ansible/roles/jenkins/Readme.md (hunk 10 above)
Comment (eemrdog) [reply user-skel/ansible/roles/jenkins/Readme.md]: ok I put a disclaimer that these are example key-value variables and depending on a use case, these names can change

File: user-skel/ansible/roles/keptn-cli/Readme.md (hunk 11 above)
Comment (eemrdog) [reply user-skel/ansible/roles/keptn-cli/Readme.md]: updated

File: user-skel/ansible/roles/microk8s/Readme.md (hunk 12 above)
Comment (eemrdog) [reply user-skel/ansible/roles/microk8s/Readme.md:4]: updated

File: user-skel/ansible/roles/repository/Readme.md (hunk 13 above)
Comment (eemrdog) [reply user-skel/ansible/roles/repository/Readme.md]: yes right, makes sense. Updated

File: user-skel/ansible/roles/monaco/Readme.md (hunk 14 above)
Comment (eemrdog) [reply user-skel/ansible/roles/monaco/Readme.md]: updated

File: user-skel/ansible/roles/dashboard/Readme.md (hunk 3 above)
Comment (eemrdog) [reply user-skel/ansible/roles/dashboard/Readme.md:86]: I added expRefs, does it look good?

File: user-skel/ansible/roles/gitlab/Readme.md (hunk 7 above)
Comment (eemrdog) [reply user-skel/ansible/roles/gitlab/Readme.md:42]: Same as gitea, Does it look good?

File: user-skel/ansible/roles/jenkins/Readme.md (hunk 10 above)
Comment (eemrdog) [reply user-skel/ansible/roles/jenkins/Readme.md]: Does it look good?

**Comment by eemrdog** [comment]
@tobigremmer-dt  Could you please check only the 4 conversations to be resolved if they look good?

File: user-skel/ansible/roles/dashboard/Readme.md [hunk 15]
```diff
@@ -0,0 +1,90 @@
...
+        url: "{{ ingress_protocol }}://simplenodeservice-simplenode-jenkins-staging.{{ ingress_domain }}"
+        - section: demo-quality-gates-jenkins
+        description: Production
+        url: "{{ ingress_protocol }}://simplenodeservice-simplenode-jenkins-production.{{ ingress_domain }}"
+        
+        guides:
+        - description: "Quality Gates, Monitoring as a Service and Monitoring as Code - Demo using Jenkins, Gitea and Cloud Automation"
+        url: "{{ ingress_protocol }}://gitea.{{ ingress_domain }}/demo/quality-gates-jenkins/src/branch/main/demo"
+
+        extRefs:
```
Comment (tobigremmer-dt) [inline user-skel/ansible/roles/dashboard/Readme.md]: Indentation is off here.

**Review by tobigremmer-dt — APPROVED** [review]
Looks great. Good job.

**Pull request merged**